server/controllers/auth.js

```diff
@@ -9,7 +9,12 @@ const generateToken = require("../utils/generateToken");
 exports.registerUser = asyncHandler(async (req, res, next) => {
   const { username, email, password } = req.body;
 
-  await checkUserExists(email, username, res);
+  try {
+    await checkUserExists(email, username);
+  } catch (error) {
+    res.status(400);
+    throw error;
+  }
```

Comment:
Nice use of the error handler! 💯  I would say [`422 Unprocessable Entity`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) fits this better than `400` (for what it's worth, it's also what GitHub uses when the request for its API fails validation).